src/main/scala/decider/Z3ProverAPI.scala

```diff
@@ -115,19 +112,59 @@ class Z3ProverAPI(uniqueId: String,
     logfileWriter = if (Verifier.config.disableTempDirectory()) null else viper.silver.utility.Common.PrintWriter(Verifier.config.proverLogFile(uniqueId).toFile)
     ctx = new Context(Z3ProverAPI.initialOptions.asJava)
     val params = ctx.mkParams()
+
+    // When setting parameters via API, we have to remove the smt. prefix
+    def removeSmtPrefix(s: String) = {
+      if (s.startsWith("smt."))
+        s.substring(4)
+      else
+        s
+    }
+
+    val useOldVersionParams = version() <= Version("4.8.7.0")
     Z3ProverAPI.boolParams.foreach{
-      case (k, v) => params.add(k, v)
+      case (k, v) =>
```

Comment:
I can't do that because params.add is overloaded for string, int, double, ... values, and I can't call a generic method that works for all parameter types. That's the reason I split up the parameters in the first place.